webknossos-datastore/app/com/scalableminds/webknossos/datastore/models/datasource/ElementClass.scala

```diff
@@ -0,0 +1,168 @@
+      case ElementClass.float  => true
+      case ElementClass.double => true
+      case ElementClass.int8   => true
+      case ElementClass.int16  => true
+      case ElementClass.int32  => true
+      case ElementClass.int64  => true
+    }
+
+  def defaultIntensityRange(elementClass: ElementClass.Value): (Double, Double) = elementClass match {
+    case ElementClass.uint8  => (0.0, math.pow(2, 8) - 1)
+    case ElementClass.uint16 => (0.0, math.pow(2, 16) - 1)
+    case ElementClass.uint24 => (0.0, math.pow(2, 8) - 1) // Assume uint24 rgb color data
+    case ElementClass.uint32 => (0.0, math.pow(2, 32) - 1)
+    case ElementClass.float  => (-3.40282347e+38, 3.40282347e+38)
+    case ElementClass.double => (-1.7976931348623157e+308, 1.7976931348623157e+308)
+    case ElementClass.int8   => (-math.pow(2, 7), math.pow(2, 7) - 1)
+    case ElementClass.int16  => (-math.pow(2, 15), math.pow(2, 15) - 1)
+    case ElementClass.int32  => (-math.pow(2, 31), math.pow(2, 31) - 1)
+
+    // Int64 types are only supported for segmentations (which don't need to call this
+    // function as there will be no histogram / color data). Still, for the record:
+    // The frontend only supports number in range of 2 ** 53 - 1 which is currently
+    // the maximum supported "64-bit" segment id due to JS Number limitations (frontend).
+    case ElementClass.uint64 | ElementClass.int64 => (0.0, math.pow(2, 8) - 1)
+    case _                                        => (0.0, 255.0)
+  }
+
+  def bytesPerElement(elementClass: ElementClass.Value): Int = elementClass match {
+    case ElementClass.uint8  => 1
+    case ElementClass.uint16 => 2
+    case ElementClass.uint24 => 3
+    case ElementClass.uint32 => 4
+    case ElementClass.uint64 => 8
+    case ElementClass.float  => 4
+    case ElementClass.double => 8
+    case ElementClass.int8   => 1
+    case ElementClass.int16  => 2
+    case ElementClass.int32  => 4
+    case ElementClass.int64  => 8
+  }
+
+  def fromProto(elementClassProto: ElementClassProto): ElementClass.Value =
+    elementClassProto match {
+      case ElementClassProto.uint8  => uint8
+      case ElementClassProto.uint16 => uint16
+      case ElementClassProto.uint24 => uint24
+      case ElementClassProto.uint32 => uint32
+      case ElementClassProto.uint64 => uint64
+      case ElementClassProto.int8   => int8
+      case ElementClassProto.int16  => int16
+      case ElementClassProto.int32  => int32
+      case ElementClassProto.int64  => int64
+      case ElementClassProto.Unrecognized(_) =>
+        throw new RuntimeException(s"Cannot convert ElementClassProto $elementClassProto to ElementClass")
+    }
+
+  def toProto(elementClass: ElementClass.Value): Box[ElementClassProto] =
+    elementClass match {
+      case ElementClass.uint8  => Full(ElementClassProto.uint8)
+      case ElementClass.uint16 => Full(ElementClassProto.uint16)
+      case ElementClass.uint24 => Full(ElementClassProto.uint24)
```

Comment:
And this was missing, correct?

Reply:
Yes. I thought ElementClassProto doesn’t have uint24, since it doesn’t really make sense for VolumeTracingsLayers anyway, but apparently it does have it, so I think for completeness it should also be handled here.